File: composeApp/src/commonMain/kotlin/com/checklistboteco/domain/model/Area.kt
package com.checklistboteco.domain.model

enum class Area(val displayName: String) {
    ATENDIMENTO("Atendimento"),
    COZINHA("Cozinha"),
    ESTOQUE("Estoque"),
    LIMPEZA("Limpeza");

    companion object {
        fun fromString(value: String): Area = entries.find { it.name == value } ?: ATENDIMENTO
    }
}

File: Packages/Models/Sources/Models/Area.swift
import Foundation

public enum Area: String, CaseIterable, Codable, Sendable {
  case atendimento = "ATENDIMENTO"
  case cozinha = "COZINHA"
  case estoque = "ESTOQUE"
  case limpeza = "LIMPEZA"

  public var displayName: String {
    switch self {
    case .atendimento: return "Atendimento"
    case .cozinha: return "Cozinha"
    case .estoque: return "Estoque"
    case .limpeza: return "Limpeza"
    }
  }

  public static func from(_ value: String) -> Area {
    Area(rawValue: value.uppercased()) ?? .atendimento
  }
}
